driver/compenv.ml

```diff
@@ -62,84 +62,6 @@ let first_objfiles = ref []
 let last_objfiles = ref []
 let stop_early = ref false
 
-<<<<<<< HEAD
-(* Check validity of module name *)
-let is_unit_name name =
-  try
-    if name = "" then raise Exit;
-    begin match name.[0] with
-    | 'A'..'Z' -> ()
-    | _ ->
-       raise Exit;
-    end;
-    for i = 1 to String.length name - 1 do
-      match name.[i] with
-      | 'A'..'Z' | 'a'..'z' | '0'..'9' | '_' | '\'' -> ()
-      | _ ->
-         raise Exit;
-    done;
-    true
-  with Exit -> false
-
-let check_unit_name filename name =
-  if not (is_unit_name name) then
-    Location.prerr_warning (Location.in_file filename)
-      (Warnings.Bad_module_name name)
-
-(* Compute name of module from output file name *)
-let module_of_filename inputfile outputprefix =
-  let basename = Filename.basename outputprefix in
-  let name =
-    try
-      (* For hidden files (i.e., those starting with '.'), include the initial
-         '.' in the module name rather than let it be empty. It's still not a
-         /good/ module name, but at least it's not rejected out of hand by
-         [Compilation_unit.Name.of_string]. *)
-      let pos = String.index_from basename 1 '.' in
```

Comment:
This bit is the source of the conflict.  Is some equivalent code needed in a new place (`Unit_info`, maybe) to be compatible with @lukemaurer's symbols changes?  I haven't traced this through all the way - I imagine Luke will know immediately.

Reply:
Ah—yes, we need to change `Unit_info.basename_chop_extensions` so that it calls `String.index_from` instead of `String.index`. I think it all hangs together besides that (thankfully, there's surprisingly little overlap between `Unit_info` and `Compilation_unit`)